### ISSUE TITLE
Setting dismissOnClickOutside to false doesn't work for top and bottom side

### ISSUE THREAD

**Issue by risalfajar** [maxkeppeler/sheets-compose-dialogs#16]
I'm trying to create a permanent/mandatory dialog using `CoreDialog`

```kotlin
val dialogState = rememberSheetState(onDismissRequest = { Timber.w("DISMISSED") })

CoreDialog(
    state = dialogState,
    selection = CoreSelection(false),
    properties = DialogProperties(dismissOnBackPress = false, dismissOnClickOutside = false),
    body = {
        Row(horizontalArrangement = Arrangement.Center, verticalAlignment = Alignment.CenterVertically, modifier = Modifier.fillMaxWidth()) {
            CircularProgressIndicator()
            Spacer(Modifier.width(16.dp))
            Text(text = "Loading...", style = MaterialTheme.typography.body2)
        }
    }
)
```

Setting `dismissOnBackPress` to `false` works as intended. But setting `dismissOnClickOutside` to `false` doesn't work fully, the dialog can be closed by clicking outside and `onDismissRequest` is called **EXCEPT** if you click on the right/left side of the dialog, then it won't be dismissed.

A problem with dialog height, maybe?

**Comment by risalfajar** [comment]
I think the problem is because onClick in `DialogBase.kt` doesn't check the dialog properties:

```kotlin
Box(
    contentAlignment = Alignment.Center,
    modifier = Modifier
        .testTag(TestTags.DIALOG_BASE_CONTAINER)
        .fillMaxSize()
        .clickable(
            interactionSource = boxInteractionSource,
            indication = null,
            onClick = state::dismiss
        )
) 
```

**Comment by maxkeppeler** [comment]
Thanks for the detailed issue. The Box, for the quick fix mentioned in the class, currently misses an additional check on the dialog properties before dismissing the dialog, or not. 

Fix for dialog: https://issuetracker.google.com/issues/221643630?pli=1
There does not seem to exist an update to that, so I keep that current fix and just add the additional check.

**Comment by risalfajar** [comment]
Awesome, thanks! Any planned date for the next version release?

**Comment by maxkeppeler** [comment]
Not exactly sure. Either, I will release 1.0.X that includes this bug fix alongside others, or I will release 1.1.0 with the landscape support. Depends on my time. If the landscape support takes much longer, I will just release 1.0.X asap.
